tests/peers_test.go

```diff
@@ -0,0 +1,88 @@
+package tests
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/log"
+	"github.com/ethereum/go-ethereum/p2p"
+	"github.com/ethereum/go-ethereum/rpc"
+	"github.com/ethereum/go-ethereum/whisper/whisperv6"
+	"github.com/status-im/status-scale/cluster"
+	"github.com/status-im/status-scale/network"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestSentEnvelope(t *testing.T) {
+	// Setup cluster
+	c := ClusterFromConfig()
+	require.NoError(t, c.Create(context.TODO(), cluster.ScaleOpts{Boot: 1, Relay: 2, Deploy: true}))
+	defer c.Clean(context.TODO()) // handle interrupt signal
+
+	var client *rpc.Client
+	// Get the rpc client
+	Eventually(t, func() error {
+		client = c.GetRelay(0).Admin().Client()
+		if client == nil {
+			return errors.New("couldn't get the rpc client")
+		}
+		return nil
+	}, 30*time.Second, 1*time.Second)
+
+	var peers []*p2p.PeerInfo
+	// Check connected peers
+	Eventually(t, func() error {
+		var err error
+		peers, err = c.GetRelay(0).Admin().Peers(context.TODO())
+		log.Trace("waiting for 1 peers", "peers", len(peers), "err", err)
+		if err != nil {
+			return err
+		}
+		if len(peers) != 1 {
+			return fmt.Errorf("peers %+v expected to be %d", peersToIPs(peers), 2)
+		}
+		return nil
+	}, 30*time.Second, 1*time.Second)
+
+	var hash common.Hash
+	msg := whisperMsg(t, client)
+
+	// With default network capabilities we should be able to succesfully send
+	// messages
+	assert.NoError(t, client.Call(&hash, "debug_postSync", msg))
+
+	// Switching connected peers network connection
+	require.NoError(t, c.GetRelay(0).EnableConditions(context.TODO(), network.Options{
+		PacketLoss: 100,
+		TargetAddr: strings.Split(peers[0].Network.RemoteAddress, ":")[0],
+	}))
+	require.NoError(t, c.GetRelay(1).EnableConditions(context.TODO(), network.Options{
+		PacketLoss: 100,
+		TargetAddr: strings.Split(peers[0].Network.RemoteAddress, ":")[0],
+	}))
+
+	// As our current node doesn't have any connected peer, no message should be sent
+	assert.Error(t, client.Call(&hash, "debug_postSync", msg))
+}
+
+func whisperMsg(t *testing.T, client *rpc.Client) whisperv6.NewMessage {
```

Comment:
i suggested above that it makes sense to make a wrapper similar to Admin. this wrapper will have 2 methods for now

```
NewSymKey(context.Context) (string, err)
PostDebug(context.Context, whisperv6.Message) (hash, error)
```